app/components/forms/events/view/create-discount-code.js

```diff
@@ -12,6 +12,16 @@ export default Component.extend(FormMixin, {
     $.fn.form.settings.rules.checkMaxTotal = () => {
       return this.get('data.maxQuantity') <= this.get('data.ticketsNumber');
     };
+    $.fn.form.settings.rules.checkTicketSelected = () => {
+      let tickets = this.eventTickets;
+      let checkedTickets = 0;
+      tickets.forEach(ticket => {
+        if (ticket.get('isChecked') === true) {
+          checkedTickets += 1;
```

Comment:
Since we only need to see if at least one ticket is selected, counting tickets is redundant. Please change the logic so that if we see even a single checked ticket we return true.

Reply:
@pradeepgangwar semantic ui doesn't provide any validation rule for checkboxes. I didn't find of any other way than to check if atleast one ticket is selected. The same is being done for toggling all tickets.
Can you suggest of some other way? :smile:

Reply:
@pradeepgangwar I have made the requsted changes.